java-frontend/src/main/java/org/sonar/java/xml/maven/PomParser.java

```diff
@@ -68,6 +69,22 @@ public static MavenProject parseXML(File file) {
     return null;
   }
 
+  /**
+   * By default, the location of XML element is enabled, except on IBM JVM where it is disabled and has to be manually enabled.
+   * The property is a IBM-specific property, not recognized by non-IBM JVMs, and will throw an {@link IllegalArgumentException}.
+   *
+   * See {@link https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.zos.80.doc/user/xml/xlxpj_reference.html}.
+   *
+   * @param factory
+   */
+  private static void enableLocationPropertyForIBM(XMLInputFactory factory) {
+    try {
+      factory.setProperty("javax.xml.stream.isSupportingLocationCoordinates", true);
+    } catch (IllegalArgumentException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)